src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -122,6 +142,28 @@ protected ExternalWorkspace run() throws Exception {
     }
 
     /**
+     * Gets the environment variables based on the current run.
+     *
+     * @return the {@link EnvVars} for the current run
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private EnvVars getEnvVars() throws IOException, InterruptedException {
```

Comment:
🐛 This code is misleading, because EnvVars are not only about Build Parameters.
IMHO you should use <code>Run#getEnvironment(TaskListener</code>) instead of this home-made method